ident/ident.go

```diff
@@ -0,0 +1,75 @@
+package ident
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"sort"
+)
+
+type Identifiable interface {
+	Identity() []byte
+}
+
+func ContentAddress(entity Identifiable) string {
+	h := sha256.New()
+	_, _ = h.Write(entity.Identity())
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+type Buffer struct {
+	buf []byte
+}
+
+func NewBuffer() *Buffer {
+	return &Buffer{
+		buf: make([]byte, 0),
+	}
+}
+
+func (b *Buffer) MarshalBytes(v []byte) {
```

Comment:
Here and in every function below, you should also stream a type byte into the hash/buffer.  Otherwise different object formats can end up with the same hash, which is a bit surprising.

Reply:
good idea, changed.